Libraries/LLM/KVCache.swift

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2024 Apple Inc.
+
+import Foundation
+import MLX
+
+public class KVCache: Evaluatable {
```

Comment:
The KVCache.  I will add the rotating cache as well, but the default `max_kv_size` is `None` so we only need this for the performance tuning.